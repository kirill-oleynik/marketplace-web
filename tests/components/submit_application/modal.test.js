const React = require('react');
const { shallow } = require('enzyme');
const toJSON = require('enzyme-to-json').default;
const SubmitApplicationModal = require('../../../app/components/submit_application/modal').default;

describe('#render', () => {
  describe('when modal is closed', () => {
    test('it renders correctly', () => {
      const component = shallow(
        <SubmitApplicationModal
          isOpen={false}
          closeModal={() => {}}
        />
      );

      expect(toJSON(component)).toMatchSnapshot();
    });
  });

  describe('when modal is opend', () => {
    test('it renders correctly', () => {
      const component = shallow(
        <SubmitApplicationModal
          isOpen={true}
          closeModal={() => {}}
        />
      );

      expect(toJSON(component)).toMatchSnapshot();
    });
  });
});
