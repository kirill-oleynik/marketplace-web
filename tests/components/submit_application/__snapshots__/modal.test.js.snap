// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render when modal is closed it renders correctly 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransitionTimeout={150}
  className="main-modal"
  fade={true}
  isOpen={false}
  keyboard={true}
  modalTransitionTimeout={300}
  zIndex={1050}
>
  <Container
    tag="div"
  >
    <div
      className="text-right pt-30"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <i
          className="icon icon-cross font-16 in-black-035"
        />
      </button>
    </div>
    <Row
      className="main-modal__content"
      tag="div"
    >
      <Col
        md={
          Object {
            "offset": 4,
            "size": 4,
          }
        }
        sm={
          Object {
            "offset": 2,
            "size": 8,
          }
        }
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="12"
      >
        <Connect(Translate(SubmitApplicationFormContainer)) />
      </Col>
    </Row>
  </Container>
</Modal>
`;

exports[`#render when modal is opend it renders correctly 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransitionTimeout={150}
  className="main-modal"
  fade={true}
  isOpen={true}
  keyboard={true}
  modalTransitionTimeout={300}
  zIndex={1050}
>
  <Container
    tag="div"
  >
    <div
      className="text-right pt-30"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <i
          className="icon icon-cross font-16 in-black-035"
        />
      </button>
    </div>
    <Row
      className="main-modal__content"
      tag="div"
    >
      <Col
        md={
          Object {
            "offset": 4,
            "size": 4,
          }
        }
        sm={
          Object {
            "offset": 2,
            "size": 8,
          }
        }
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="12"
      >
        <Connect(Translate(SubmitApplicationFormContainer)) />
      </Col>
    </Row>
  </Container>
</Modal>
`;
